packages/@ember/-internals/glimmer/lib/syntax/outlet.ts

```diff
@@ -66,87 +71,48 @@ export function outletHelper(args: VMArguments, vm: VM) {
   let nameRef: Reference<string>;
 
   if (args.positional.length === 0) {
-    nameRef = new ConstReference('main');
+    nameRef = createPrimitiveRef('main');
   } else {
-    nameRef = args.positional.at<PathReference<string>>(0);
+    nameRef = args.positional.at(0);
   }
 
-  return new OutletComponentReference(new OutletReference(scope.outletState, nameRef));
-}
+  let outletRef = createComputeRef(() => {
+    let state = valueForRef(scope.get('outletState'));
+    let outlets = state !== undefined ? state.outlets : undefined;
 
-class OutletModelReference extends RootReference {
-  constructor(private parent: PathReference<OutletState | undefined>) {
-    super();
+    return outlets !== undefined ? outlets[valueForRef(nameRef)] : undefined;
+  });
 
-    if (DEBUG) {
-      this.debugLabel = '@model';
-    }
-  }
+  let lastState: Option<OutletDefinitionState> = null;
+  let definition: Option<CurriedComponentDefinition> = null;
 
-  isConst() {
-    return false;
-  }
+  return createComputeRef(() => {
+    let state = stateFor(outletRef);
 
-  compute(): unknown {
-    let state = this.parent.value();
+    if (!validate(state, lastState)) {
+      lastState = state;
 
-    if (state === undefined) {
-      return undefined;
-    }
+      if (state !== null) {
+        let named = dict<Reference>();
+        named.model = childRefFromParts(outletRef, ['render', 'model']);
 
-    let { render } = state;
+        if (DEBUG) {
+          named.model.debugLabel = '@model';
```

Comment:
Should this use the `createAliasReference` (or wahtever it is called)?

Reply:
Yup, that works!